test/spmd/tensor/parallel/test_parallelize_api.py

```diff
@@ -0,0 +1,121 @@
+# Owner(s): ["oncall: distributed"]
+
+import torch
+from torch.testing._internal.common_utils import run_tests
+from spmd.testing.common_dtensor import DTensorTestBase, with_comms
+from spmd.tensor import distribute_tensor, DeviceMesh, Shard, Replicate
+from spmd.tensor.parallel import PairwiseParallel, ParallelStyle
+from spmd.tensor.parallel.api import _parallelize_mlp
+from spmd.tensor.parallel.utils import _create_1d_device_mesh
+from spmd.tensor.parallel.style import (
+    make_input_replicate_1d,
+    make_output_replicate_1d,
+)
+
+
+class MLPModule(torch.nn.Module):
+    def __init__(self, device):
+        super(MLPModule, self).__init__()
+        torch.manual_seed(5)
+        self.net1 = torch.nn.Linear(10, 16, device=device)
+        self.relu = torch.nn.ReLU()
+        self.net2 = torch.nn.Linear(16, 12, device=device)
+
+    def forward(self, x):
+        return self.net2(self.relu(self.net1(x)))
+
+
+class TensorParallelAPITests(DTensorTestBase):
+    @with_comms
+    def test_creat_1d_device_mesh(self):
+        tp_size = 2
+        mesh_shape = (
+            torch.arange(self.world_size)
+            .reshape(
+                self.world_size // (self.world_size // tp_size),
+                self.world_size // tp_size,
+            )
+            .to(torch.int)
+        )
+        mesh = DeviceMesh(self.device_type, mesh_shape)
+        one_dimention_mesh_shape = mesh_shape[self.rank // tp_size, :]
+        pg = mesh.get_dim_groups()[1]
+        new_mesh = _create_1d_device_mesh(mesh, 1)
```

Comment:
can we also add a case to test tp_dim=0?

Reply:
Sure.